.github/workflows/CI.yml

```diff
@@ -7,161 +7,125 @@ on:
-          override: true
-      - name: Patch
-        run: |
-          set -e
-          # Remove any existing patch statements
-          mv Cargo.toml Cargo.toml.bck
-          sed -n '/\[patch.crates-io\]/q;p' Cargo.toml.bck > Cargo.toml
-          # Patch all crates
-          cat .github/workflows/patch.toml >> Cargo.toml
-          # Print `Cargo.toml` for debugging
-          echo "~~~~ Cargo.toml ~~~~"
-          cat Cargo.toml
-          echo "~~~~~~~~~~~~~~~~~~~~"
-      - name: Build
-        uses: actions-rs/cargo@v1
-        with:
-          command: check
-          args: -p ${{ matrix.crate }} --verbose
-      - name: Run tests
-        uses: actions-rs/cargo@v1
-        env:
-          CI: 'True'
-          # RUSTFLAGS: '-D warnings'
-        with:
-          command: test
-          args: -p ${{ matrix.crate }} --verbose --all-features
-
-  features:
-
+    - uses: actions/checkout@master
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: ${{ matrix.rust }}
+        profile: minimal
+    - name: Run tests
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --all --all-features
+
+  style:
+    # Check style.
+    needs: check
     runs-on: ubuntu-latest
-
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v1
-      - name: Install rust toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: nightly
-          override: true
-      - name: Patch
-        run: |
-          set -e
-          # Remove any existing patch statements
-          mv Cargo.toml Cargo.toml.bck
-          sed -n '/\[patch.crates-io\]/q;p' Cargo.toml.bck > Cargo.toml
-          # Patch all crates
-          cat .github/workflows/patch.toml >> Cargo.toml
-          # Print `Cargo.toml` for debugging
-          echo "~~~~ Cargo.toml ~~~~"
-          cat Cargo.toml
-          echo "~~~~~~~~~~~~~~~~~~~~"
-      - name: Install cargo-hack
-        uses: actions-rs/cargo@v1
-        with:
-          command: install
-          args: cargo-hack
-      - name: Run feature checks
-        uses: actions-rs/cargo@v1
-        env:
-          CI: 'True'
-        with:
-          command: hack
-          # Check each feature works properly
-          # * --each-feature
-          #   run for each feature which includes --no-default-features and default features of package
-          # * -Z avoid-dev-deps
-          #   build without dev-dependencies to avoid https://github.com/rust-lang/cargo/issues/4866
-          #   tracking-issue: https://github.com/rust-lang/cargo/issues/5133
-          args: check -p tower --each-feature -Z avoid-dev-deps
-
-  rustfmt:
-
+    - uses: actions/checkout@master
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        components: rustfmt
+        profile: minimal
+    - name: rustfmt
+      uses: actions-rs/cargo@v1
+      with:
+        command: fmt
+        args: --all -- --check
```

Comment:
Should we also run a clippy check?

Reply:
There is but we have a lot of warnings so I am going to try and fix that in another PR.